### PR TITLE
Prevent package splitting by not sending packages larger than UDP size

### PR DESCRIPTION
**Pull request by grosser**
fixes https://github.com/DataDog/dogstatsd-ruby/issues/81

 - deprecated argument after options introduced in https://github.com/DataDog/dogstatsd-ruby/commit/d910ba9e8bf5214df5b0a573c8f833e504fa21c4
 - flush buffer if adding the current message would make the udp package too large

@masci @jonmoter

### CODE REVIEW

File: lib/datadog/statsd.rb [hunk 1]
```diff
@@ -90,15 +95,20 @@ def self.VERSION
...
       @prefix = nil
       @socket = connect_to_socket if @socket_path.nil?
       self.namespace = opts[:namespace]
       self.tags = opts[:tags]
-      @buffer = Array.new
-      self.max_buffer_size = max_buffer_size
+
+      # buffering
+      @max_buffer_size = opts[:max_buffer_size] || bad_old_max_buffer_size
+      @buffer = String.new
```
Comment (grosser) [inline lib/datadog/statsd.rb]: not using `""` to avoid it being frozen when we switch to frozen strings

File: lib/datadog/statsd.rb [hunk 2]
```diff
@@ -73,7 +78,7 @@ class Statsd
     attr_reader :buffer
 
     # Maximum number of metrics in the buffer before it is flushed
-    attr_accessor :max_buffer_size
+    attr_reader :max_buffer_size
```
Comment (gmmeyer) [inline lib/datadog/statsd.rb]: 🍰

File: lib/datadog/statsd.rb (hunk 1 above)
Comment (gmmeyer) [reply lib/datadog/statsd.rb]: Can you add this in a comment?

Comment (grosser) [reply lib/datadog/statsd.rb]: added

**Comment by grosser** [comment]
needs to be rebased on https://github.com/DataDog/dogstatsd-ruby/pull/83

**Pull request closed without merging**